### ISSUE TITLE
Error when checking target: expected activation_2 to have shape (10,) but got array with shape (1,)

### ISSUE THREAD

**Issue by lamhoson** [keras-team/keras#11904]
Please make sure that the boxes below are checked before you submit your issue.
If your issue is an **implementation question**, please ask your question on [StackOverflow](http://stackoverflow.com/questions/tagged/keras) or [on the Keras Slack channel](https://keras-slack-autojoin.herokuapp.com/) instead of opening a GitHub issue.

Thank you!

- [ ] Check that you are up-to-date with the master branch of Keras. You can update with:
`pip install git+git://github.com/keras-team/keras.git --upgrade --no-deps`

- [ ] Check that your version of TensorFlow is up-to-date. The installation instructions can be found [here](https://www.tensorflow.org/get_started/os_setup).

- [ ] Provide a link to a GitHub Gist of a Python script that can reproduce your issue (or just copy the script here if it is short).

**Comment by lamhoson** [comment]
Anyone know what are wrong in following's TensorFlow's Keras Model  ?

model = tf.keras.Sequential()
model.add(layers.Conv2D(NB_FILTER, kernel_size=KERNAL_SIZE, padding='same',
                        input_shape=(rows, columns, CHANNELS))); model.add(layers.Activation('relu'))
model.add(layers.MaxPooling2D(pool_size=POOL_SIZE))
model.add(layers.Dropout(0.25))
 
model.add(layers.Conv2D(NB_FILTER*2, kernel_size=KERNAL_SIZE, strides=STRIDES)); model.add(layers.Activation('relu'))
model.add(layers.MaxPooling2D(pool_size=POOL_SIZE))
model.add(layers.Dropout(0.25))
 
model.add(layers.Flatten())
model.add(layers.Dense(64, activation='relu')) #try from 64,128,256,512. O?P layer=Relu. A.9.
model.add(layers.Dropout(0.5))
model.add(layers.Dense(10)) # 10=Number of classes to be classified, O?P layer=softmax
model.add(layers.Activation('softmax'))

********* The constants
NB_FILTER =32 # number of trainable features filters, so-called convolution but it is correlation in fact
KERNAL_SIZE =(3,3) # convolution kernel size 3x3.
POOL_SIZE =(2, 2) # size of pooling area for max pooling.
STRIDES=(3,3) # pooling filter pixel move step, 3=move 3 pixels. For dimensional reduction
NB_EPOCHS=3 # number of repeative trainings. 5 means 5 loops. Having 5 times of weight updates
SPILT_RATIO_4T=0.2  # dataset spilt ratio for test set, reminded for train set
CHANNELS = 1 # image channel is one for gray image


![model](https://user-images.githubusercontent.com/34258914/50208529-5ceffa00-03ac-11e9-9b10-4f7bdc953385.png)
![model_summary](https://user-images.githubusercontent.com/34258914/50208623-9a548780-03ac-11e9-8704-f15623dbf6ca.PNG)

**Comment by Golbstein** [comment]
Nothing.

```
**from keras.models import Sequential**
model = Sequential()
model.add(layers.Conv2D(NB_FILTER, kernel_size=KERNAL_SIZE, padding='same',
input_shape=(224, 224, 3))); model.add(layers.Activation('relu'))
model.add(layers.MaxPooling2D(pool_size=POOL_SIZE))
model.add(layers.Dropout(0.25))

model.add(layers.Conv2D(NB_FILTER*2, kernel_size=KERNAL_SIZE, strides=STRIDES)); model.add(layers.Activation('relu'))
model.add(layers.MaxPooling2D(pool_size=POOL_SIZE))
model.add(layers.Dropout(0.25))

model.add(layers.Flatten())
model.add(layers.Dense(64, activation='relu')) #try from 64,128,256,512. O?P layer=Relu. A.9.
model.add(layers.Dropout(0.5))
model.add(layers.Dense(10)) # 10=Number of classes to be classified, O?P layer=softmax
model.add(layers.Activation('softmax'))
```

**Comment by lamhoson** [comment]
Thank a lot.  It run after adjust the label data as followings:
label_train = to_categorical(label_train, NB_CLASSES)
label_test = to_categorical(label_test, NB_CLASSES)




===================
- The labels' shape is: labels=labels.reshape(nbImages,1)
- Any reference that i can understand deeper what to_categorical() is actually doing for me.




Hoson
On 12/19/2018 20:08，Jenia Golbstein<notifications@github.com> wrote：

Nothing.

**from keras.models import Sequential**
model = Sequential()
model.add(layers.Conv2D(NB_FILTER, kernel_size=KERNAL_SIZE, padding='same',
input_shape=(224, 224, 3))); model.add(layers.Activation('relu'))
model.add(layers.MaxPooling2D(pool_size=POOL_SIZE))
model.add(layers.Dropout(0.25))

model.add(layers.Conv2D(NB_FILTER*2, kernel_size=KERNAL_SIZE, strides=STRIDES)); model.add(layers.Activation('relu'))
model.add(layers.MaxPooling2D(pool_size=POOL_SIZE))
model.add(layers.Dropout(0.25))

model.add(layers.Flatten())
model.add(layers.Dense(64, activation='relu')) #try from 64,128,256,512. O?P layer=Relu. A.9.
model.add(layers.Dropout(0.5))
model.add(layers.Dense(10)) # 10=Number of classes to be classified, O?P layer=softmax
model.add(layers.Activation('softmax'))

**Comment by msymp** [comment]
This issue seems to be resolved as confirmed by @lamhoson , hence we close this. Thanks.
